src/layers/schematicInterfaces.ts

```diff
@@ -105,6 +106,124 @@ export interface Cement {
   toc: number;
 }
 
+export type PerforationSubKind =
+  | 'Perforation'
+  | 'Open hole gravel pack'
+  | 'Open hole screen'
+  | 'Open hole'
+  | 'Open hole frac pack'
+  | 'Cased hole frac pack'
+  | 'Cased hole gravel pack'
+  | 'Cased hole fracturation';
+
+export interface Perforation {
+  kind: 'perforation';
+  subKind: PerforationSubKind;
+  id: string;
+  top: number;
+  bottom: number;
+  isOpen: boolean;
+  casingIds: string[];
+  holeId?: string[];
```

Comment:
Maybe we can combine the reference ids into one array called `referenceIds`, like in https://github.com/equinor/esv-intersection/pull/515?
The ids will reference the ids of each hole/casing dimension that we pass through to the intersection.
E.g. full of `id`s instead of parent ids like `holeId` and `casingId`.